drivers/hmis/app/models/hmis/filter/enrollment_filter.rb

```diff
@@ -49,4 +50,35 @@ def with_project_types(scope)
   def with_search_term(scope)
     with_filter(scope, :search_term) { scope.matching_search_term(input.search_term) }
   end
+
+  def with_household_tasks(scope)
+    with_filter(scope, :household_tasks) do
+      if input.household_tasks&.include?('ANNUAL_DUE')
+        entry_anniversary = Arel.sql('make_date(extract(year from current_date)::integer, extract(month from "EntryDate")::integer, extract(day from "EntryDate")::integer)')
+        start_date = Arel.sql('make_date(extract(year from current_date)::integer, extract(month from "EntryDate")::integer, extract(day from "EntryDate")::integer) - interval \'30 days\'')
+
+        assessment_ids = Hmis::Hud::CustomAssessment.
+          where(
+            enrollment_id: scope.pluck(:enrollment_id),
+            data_source_id: scope.joins(:project).pluck(p_t[:data_source_id]).uniq,
+            data_collection_stage: 5,
+            wip: false,
+          ).
+          pluck(:id)
+
+        # SQL-ized version of the logic here: drivers/hmis/app/models/hmis/reminders/reminder_generator.rb#annual_assessment_reminder
+        scope = scope.
+          left_outer_joins(:exit).
+          joins(e_t.join(cas_t, Arel::Nodes::OuterJoin).on(cas_t[:enrollment_id].eq(e_t[:enrollment_id]).and(cas_t[:id]).in(assessment_ids)).join_sources).
+          # Include anything where the entry date is outside of the year window
+          where(e_t[:entry_date].lteq(Date.today - 1.year + 30.days)).
+          # Include anything that isn't exited, or was exited after the entry anniversary
+          where(ex_t[:exit_date].eq(nil).or(ex_t[:exit_date].gt(entry_anniversary))).
```

Comment:
I might be missing something but it seems like this would use the anniversary of the enrollment entry date. But I think it should be the entry date of the HOH, which could be different. Perhaps some inaccuracy is okay?

More broadly, I wonder if there's a way we can avoid this SQL implementation. Maybe we could pre calculate the assessment due dates and then this query becomes simpler. But that's a design question for when Gig returns

Reply:
I concur, the anniversary date should be x-years from the HoH Entry Date.

Maybe instead of pre-calculating, we could make a helper class that takes some columns, etc. and returns the SQL for the anniversary range.  At least that way we could standardize how these are calculated (I think we need to do that in a handful of places) and if we decided to change it, it would be centralized so should be easy to update as long as we can keep the same API.

Reply:
> the anniversary date should be x-years from the HoH Entry Date.

Per standup convo 11/13, this is actually the earliest entry date for any enrollment in the household, not the current HOH entry date

Reply:
From the Glossary:
TLDR (calculate from the earliest enrollment start date in the household)
```
Annual Assessment
Reports utilizing Annual Assessment ([data collection stage] = 5) data on stayers in the project 365 days or more require data from the specific Annual Assessment on the head of household’s anniversary most relevant to the [report date range]. The instructions below describe how to select this exact record for each client.
1.	Use the days between the earliest Project Start Date of the household’s project enrollment and the end date of the report when determining length of stay for this purpose. 
2.	In the event a household has more than one head of household active in the report range, (i.e., if the first HoH exited and another household member became the HoH), the later HoH’s [project start date] will be back-dated to the first HoH’s [project start date] and the first HoH’s [relationship to head of household] will be updated to something other than “Self”. This will maintain determining the Annual Assessment timing on the earliest [project start date] for any given household.
3.	Calculate the head of household’s number of years in the project. This can be done using the same algorithm as for calculating a client’s age as of a certain date. Use the client’s [project start date] and the [report end date] as the two dates of comparison. This calculation is time-based, not service-based, so a client with a [project start date] that is active for a year or longer in a night-by-night emergency shelter would require an Annual Assessment, regardless of how many bed night dates were recorded. It is important to use the “age” method of determining client anniversaries due to leap years; using “one year = 365 days” will eventually incorrectly offset the calculated anniversaries of long-term stayers.
4.	If the HoH’s number of years in the project is 0, the household is not yet required to have an annual assessment. 
5.	If the HoH’s number of years in the project is more than 0, add that number to the year of the HoH’s [project start date]. This becomes the household’s relevant anniversary date for the purposes of the report. For example, using a report date range of 10/1/2022 – 9/30/2023, a HoH with a [project start date] of 6/1/2021 will have been in the project 2 years as of 9/30/2023 and so will have an [anniversary date] of 6/1/2023. (i.e. 2021 + 2 = 2023). 
6.	Use the latest annual assessment ([data collection stage] = 5) for each client in the household dated between:
a.	30 days prior to the [anniversary date] (even if this date falls before the [report start date]) 
b.	and the lesser of (30 days after the [anniversary date], [report end date])
7.	If the household member’s [project start date] is > [anniversary date] (i.e., the person entered the household after the annual assessment may have been collected), that person is not yet required to have an annual assessment.
8.	Exclude any data with an [information date] > [report end date], even though it is legitimate for a client’s relevant annual assessment data to fall in this date range for clients whose anniversaries are near to the end of the [report date range].
9.	Clients with no annual assessment data in the relevant date range as indicated in step 6 may be reported as “missing annual assessment” or may be completely omitted depending on the report question.
10.	Clients who aren’t required to have an annual assessment yet as determined in steps 4 and 7 may be reported accordingly or may be completely omitted depending on the question.
```

Reply:
@eanders @ttoomey I fixed this up to both be cleaner and match the spec better. Take a look and let me know if you'd like any more improvements. I also added tests for this, please let me know if you'd like more test cases in there.

Reply:
I think I'm good.  I'll defer to Gig for a thorough review of that, on the surface, it looks correct to me.